docs/developer-guide/nodejs-api.md

```diff
@@ -397,385 +425,357 @@ var cli = new CLIEngine({
-    processors: {
-        ".txt": {
-            preprocess: function(text) {
-                return [text];
-            },
-            postprocess: function(messages) {
-                return messages[0];
-            }
-        }
+
+const results = linter.verify("// some source text", {
+    rules: {
+        "my-custom-rule": "error",
+        "another-custom-rule": "warn"
     }
 });
 ```
 
-### CLIEngine#isPathIgnored()
+### Linter#getRules
 
-Checks if a given path is ignored by ESLint.
+This method returns a map of all loaded rules.
 
 ```js
-var CLIEngine = require("eslint").CLIEngine;
+const Linter = require("eslint").Linter;
+const linter = new Linter();
 
-var cli = new CLIEngine({
-    ignore: true,
-    ignorePath: ".customIgnoreFile"
-});
+linter.getRules();
 
-var isIgnored = cli.isPathIgnored("foo/bar.js");
+/*
+Map {
+  'accessor-pairs' => { meta: { docs: [Object], schema: [Array] }, create: [Function: create] },
+  'array-bracket-newline' => { meta: { docs: [Object], schema: [Array] }, create: [Function: create] },
+  ...
+}
+*/
 ```
 
-### CLIEngine#getFormatter()
-
-Retrieves a formatter, which you can then use to format a report object. The argument is either the name of a built-in formatter:
+### Linter#defineParser
 
-* "[checkstyle](../user-guide/formatters#checkstyle)"
-* "[codeframe](../user-guide/formatters#codeframe)"
-* "[compact](../user-guide/formatters#compact)"
-* "[html](../user-guide/formatters#html)"
-* "[jslint-xml](../user-guide/formatters#jslint-xml)"
-* "[json](../user-guide/formatters#json)"
-* "[junit](../user-guide/formatters#junit)"
-* "[stylish](../user-guide/formatters#stylish)" (the default)
-* "[table](../user-guide/formatters#table)"
-* "[tap](../user-guide/formatters#tap)"
-* "[unix](../user-guide/formatters#unix)"
+Each instance of `Linter` holds a map of custom parsers. If you want to define a parser programmatically you can add this function
+with the name of the parser as first argument and the [parser object](/docs/developer-guide/working-with-plugins.md#working-with-custom-parsers) as second argument.
 
-or the full path to a JavaScript file containing a custom formatter. You can also omit the argument to retrieve the default formatter.
+If during linting the parser is not found, it will fallback to `require(parserId)`.
 
 ```js
-var CLIEngine = require("eslint").CLIEngine;
+const Linter = require("eslint").Linter;
+const linter = new Linter();
 
-var cli = new CLIEngine({
-    envs: ["browser", "mocha"],
-    useEslintrc: false,
-    rules: {
-        semi: 2
+linter.defineParser("my-custom-parser", {
+    parse(code, options) {
+        // ...
     }
 });
 
-// lint myfile.js and all files in lib/
-var report = cli.executeOnFiles(["myfile.js", "lib/"]);
+const results = linter.verify("// some source text", { parser: "my-custom-parser" });
+```
 
-// get the default formatter
-var formatter = cli.getFormatter();
+### Linter#version/Linter.version
```

Comment:
the link seems broken. I think It might be broken before the change. :(

Reply:
Wouldn't surprise me. I can dig in and try to fix, I suppose.